src/shell/interpreter.zig

```diff
@@ -2121,6 +2121,13 @@ pub const Interpreter = struct {
                     assert(this.child_state == .cmd_subst);
                 }
 
+                if (exit_code != 0 and
```

Comment:
When does this happen? I think a comment showing an example where this case would happen would be helpful